src/main/java/com/vinylteam/vinyl/web/servlets/OneVinylOffersServlet.java

```diff
@@ -62,47 +65,68 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
                     uniqueVinyl.getRelease(), uniqueVinyl.getFullName());
             attributes.put("discogsLink", discogsLink);
         } catch (ParseException e) {
-            e.printStackTrace();
+            log.error("discogsService.getDiscogsLink() thrown ParseException", e);
         }
 
-        List<OneVinylOffersServletResponse> offersResponseList = new ArrayList<>();
         for (Offer offer : offers) {
-            for (Shop shop : shopsFromOffers) {
-                if (offer.getShopId() == shop.getId()) {
-                    OneVinylOffersServletResponse offersResponse = new OneVinylOffersServletResponse();
-                    offersResponse.setPrice(offer.getPrice());
-                    if (offer.getCurrency().isPresent()) {
-                        offersResponse.setCurrency(offer.getCurrency().get().getSymbol());
-                    } else {
-                        offersResponse.setCurrency("");
-                    }
-                    offersResponse.setCatNumber(offer.getCatNumber());
-                    offersResponse.setInStock(offer.isInStock());
-                    offersResponse.setOfferLink(offer.getOfferLink());
-                    offersResponse.setShopImageLink(shop.getSmallImageLink());
+            var offerShopParser = parserHolder.getShopParser(offer.getShopId());
+            var shop = shopsFromOffers.stream().filter(store -> store.getId() == offer.getShopId()).findFirst().get();
+            if (offerShopParser.isPresent()) {
+                VinylParser shopParser = offerShopParser.get();
+
+                var dynamicOffer = shopParser.getRawOfferFromOfferLink(offer.getOfferLink());
+                offerService.mergeOfferChanges(offer, shopParser, dynamicOffer);
+
+                if (offer.isInStock()) {
+                    OneVinylOffersServletResponse offersResponse = getOfferResponseFromOffer(offer, shop);
```

Comment:
We show offers that are in stock.
We don't show offer when dynamicOffer isn't valid.